src/Core/Parse.hs

```diff
@@ -46,7 +46,7 @@ type ParseInlines = Gamma -> Error [InlineDef]
 parseCore :: FilePath -> IO (Error (Core, ParseInlines))
 parseCore fname
   = do input <- readInput fname
-       return (lexParse False (requalify . allowDotIds) program fname 1 input)
+       return $ ignoreSyntaxWarnings $ lexParse False (requalify . allowDotIds) program fname 1 input
```

Comment:
lexParse is changed to add syntax warnings to it's state, and report them in batch after parsing. We assume core ifaces don't have syntax warnings.